streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -242,8 +250,9 @@ public void addOwnedPartitions(final Collection<TopicPartition> ownedPartitions,
         }
     }
 
-    public void addPreviousTasksAndOffsetSums(final Map<TaskId, Long> taskOffsetSums) {
+    public void addPreviousTasksAndOffsetSums(final String consumerId, final Map<TaskId, Long> taskOffsetSums) {
         this.taskOffsetSums.putAll(taskOffsetSums);
+        consumerToPreviousTaskIds.put(consumerId, taskOffsetSums.keySet());
```

Comment:
We have several new methods, and also this new book-kept collection (`consumerToPreviousTaskIds`), but no new tests for them in ClientStateTest. Can you add the missing coverage?

The new methods are more a matter of principle; I'm really concerned that we should have good coverage on the bookkeeping aspect of `consumerToPreviousTaskIds` because I fear future regressions when we have to maintain two data structures in a consistent fashion

Reply:
Definitely. I meant to write tests but then I took Luna for a walk and forgot 😄